### PR TITLE
Initial draft for adding pyteal code snippets to docs.

### PR DESCRIPTION
**Pull request by ejbaran**
## Style notes to consider while reviewing:
1. The TEAL examples have placeholders represented with descriptive branch names and commenting. The initial approach for PyTeal was to use descriptive variable names (sometimes) and to return 1. Other considerations would be to return 0 or to return an error. The problem with returning 1 is that it is very permissive and it doesn't emphasize branching examples that well (both branches return 1). 
2. Every snippet is contained in an `approval_program` method and then followed by a write to file. The goal was to make sure that every code snippet compiles for the user (and writing to file also shows which mode to use). But for simple examples is this too much? We need to decide on a good style tenet to follow here.

### CODE REVIEW

File: docs/features/asc1/stateful/index.md [hunk 1]
```diff
@@ -43,21 +50,39 @@ Global storage for the current contract can also be modified by the smart contra
...
+```text tab="TEAL"
 byte "Mykey"
 int 50
 app_global_put
 ```
 
-To store a value in local storage, the following TEAL can be used.
+```python tab="PyTeal"
+def approval_program():
+    return(App.globalPut(Bytes("Mykey"), Int(50)))
```
Comment (jasonpaulos) [inline docs/features/asc1/stateful/index.md:61]: NIT: can you remove outer parenthesis from all the return statements?
```suggestion
    return App.globalPut(Bytes("Mykey"), Int(50))
```

**Review by jasonpaulos — COMMENTED** [review]
I've left some minor comments about style, but the PyTeal code here all looks correct and accurate. In response to the two style questions:
1. I think the existing placeholders do a good job, since they are all prefixed with a comment saying they should be changed, and some of them return 0 when it's more appropriate.
2. I think including the `approval_program` function header and code to output to a file for every PyTeal example is too much, especially because some of snippets are not standalone programs. I think these should definitely be included in the boilerplate example at the end of the doc, but I'm wondering whether something like this would work for the smaller code snippets:
```python
program = App.localPut(Int(0), Bytes("MyLocalKey"), Int(50))

print(compileTeal(program, Mode.Application))
```

**Comment by ejbaran** [comment]
Thumbs up to all your suggestions. I'll make updates.

**Comment by ejbaran** [comment]
Made updates based on @jasonpaulos suggestions. Adding Jason W for review.

**Review by JasonWeathersby — APPROVED** [review]

File: docs/features/asc1/stateful/index.md [hunk 2]
```diff
@@ -225,14 +338,33 @@ notoptingin:
...
+```text tab="TEAL"
 int 0 
 txn ApplicationID
 app_opted_in
 ```
 
+```python tab="PyTeal"
+program = App.optedIn(Int(0), Txn.application_id())
+
+print(compileTeal(approval_program(), Mode.Application))
```
Comment (jasonpaulos) [inline docs/features/asc1/stateful/index.md:365]: @ejbaran looks good, except this line still uses `approval_program()`.

Comment (ejbaran) [reply docs/features/asc1/stateful/index.md:365]: I'll patch that now! Good catch.

**Pull request merged**